src/components/Department/Department.jsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component } from 'react';
+import { Redirect } from 'react-router-dom';
+import {
+  Button,
+  ButtonGroup,
+  Col,
+  Container,
+  Row,
+} from 'reactstrap';
+import API from '../../API';
+import './Department.css';
+
+export default class Department extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      departmentInfo: null,
+      error: null,
+    };
+  }
+
+  componentDidMount() {
+    this.getDepartmentInfo();
+  }
+
+  getDepartmentInfo = async () => {
+    const { match: { params: { id } } } = this.props || null;
```

Comment:
will cause error if null

Reply:
Should I use an empty string instead of null?

Reply:
i bet you should rely on ?. there

Reply:
like this one?
 
const id = this.props?.match?.params?.id;

Reply:
yep but this.props.match?.params?.id;

Reply:
got it, thanks